core/core_modules/admin.js

```diff
@@ -0,0 +1,167 @@
+			exports.config.users[name][threadId] = [];
+		}
+
+		switch (method) {
+			case 'grant':
+			{
+				if (!exports.config.users[name][threadId].includes(action)) {
+					exports.config.users[name][threadId].push(action);
+					return true;
+				}
+				return false;
+			}
+			case 'revoke':
+			{
+				var newArr = exports.config.users[name][threadId].filter(function(item) {
+					return item !== action;
+				});
+				if (newArr.length === exports.config.users[name][threadId].length) {
+					return false;
+				}
+				exports.config.users[name][threadId] = newArr;
+				return true;
+			}
+		}
+	},
+
+	setup = function (action, name, permission) {
+		if (!exports.config.modules) {
+			exports.config.modules = {};
+		}
+
+		if (!exports.config.modules[name]) {
+			exports.config.modules[name] = [];
+		}
+
+		switch (action) {
+			case 'create':
+				{
+					if (!exports.config.modules[name].includes(permission)) {
+						exports.config.modules[name].push(permission);
+						return true;
+					}
+					return false;
+				}
+			case 'delete':
+				{
+					var newArr = exports.config.modules[name].filter(function (item) {
+						return item === action;
+					});
+					if (newArr.length === exports.config.modules[name].length) {
+						return false;
+					}
+					exports.config.modules[name] = newArr;
+					return true;
+				}
+		}
+	},
+
+	matchHook = function (moduleName, origionalMatch, config) {
+		return function(event, commandPrefix) {
+			if (getHasPermission(config, event.sender_id, event.sender_name, event.thread_id, moduleName)) {
+				return origionalMatch(event, commandPrefix);
+			}
+			return false;
+		};
+	},
+
+	helpHook = function (moduleName, origionalHelp, config) {
+		return function(commandPrefix) {
+			if (getHasPermission(config, event.sender_id, event.sender_name, event.thread_id, moduleName)) {
+				return origionalHelp.call(this, commandPrefix);
+			}
+			return false;
+		};
+	};
+
+exports.load = function () {
+    exports.config = config.getConfig('admin');
+    for (var i = 0; i < exports.platform.coreModules.length; i++) {
+        var help = exports.platform.coreModules[i].help,
+            match = exports.platform.coreModules[i].match;
+
+        exports.platform.coreModules[i].help = helpHook(exports.platform.coreModules[i].name, help, exports.config);
+        exports.platform.coreModules[i].match = matchHook(exports.platform.coreModules[i].name, match, exports.config);
+    }
+};
+
+exports.match = function (event, commandPrefix) {
+    return event.arguments[0] === commandPrefix + 'admin';
+};
+
+exports.help = function (commandPrefix) {
+    return [
+        [commandPrefix + 'admin <grant/revoke> <fullName/userId> <permissionName>', 'Grant or revoke a permission for a user.',
+            'Assigns/removes a permission from a user. Once a user has a permission they can use any core modules with the corresponding permission (and any that have no permissions/none configured).'],
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Line is too long.](https://www.codacy.com/app/theprojectshop/Kassy/file/2560344994/issues/source?bid=3282987&fileBranchId=3292072#l128)